xmake.lua of mailio: boost needs asio = true

### Xmake 版本

3.0.8

### 操作系统版本和架构

Debian 13

### 描述问题

mailio install failed because boost from mailio lost asio = true

### 期待的结果

xmake.lua of mailio: boost add asio = true

### 工程配置

add_requires("mailio")

### 附加信息和错误日志

m/mailio/0.24.1/source/include/mailio/pop3.hpp:27:10: fatal error: boost/asio.hpp: 没有那个文件或目录

--- a/packages/m/mailio/xmake.lua
+++ b/packages/m/mailio/xmake.lua
@@ -6,6 +6,7 @@ package("mailio")
     add_urls("https://github.com/karastojko/mailio/archive/refs/tags/$(version).tar.gz",
              "https://github.com/karastojko/mailio.git")
 
+    add_versions("0.25.3", "12b79d8a8211033d7e59be2e30521a8109ed83bda86c86437ebe7e04298a5aa5")
     add_versions("0.24.1", "52d5ced35b6a87677d897010fb2e7c2d2ddbd834d59aab991c65c0c6627af40f")
     add_versions("0.23.0", "9fc3f1f803a85170c2081cbbef2e301473a400683fc1dffefa2d6707598206a5")
 
@@ -14,7 +15,7 @@ package("mailio")
     end
 
     add_deps("cmake")
-    add_deps("boost", {configs = {regex = true, date_time = true, system = true, exception = true, container = true}})
+    add_deps("boost", {configs = {asio = true, system = true, exception = true}})
     add_deps("openssl")
 
     on_load(function (package)

--- a/packages/m/mailio/xmake.lua
+++ b/packages/m/mailio/xmake.lua
@@ -11,11 +11,11 @@ package("mailio")
     add_versions("0.23.0", "9fc3f1f803a85170c2081cbbef2e301473a400683fc1dffefa2d6707598206a5")
 
     if is_plat("linux", "bsd") then
-        add_syslinks("m", "pthread")
+        add_syslinks("pthread")
     end
 
     add_deps("cmake")
-    add_deps("boost", {configs = {asio = true, system = true, exception = true}})
+    add_deps("boost", {configs = {asio = true}})
     add_deps("openssl")
 
     on_load(function (package)
@@ -25,19 +25,16 @@ package("mailio")
     end)
 
     on_install("!iphoneos and !wasm", function (package)
-        local version = package:version()
         io.replace("CMakeLists.txt", "/WX", "", {plain = true})
         io.replace("CMakeLists.txt", "set(Boost_USE_STATIC_LIBS ON)", "", {plain = true})
 
-        if package:gitref() or version:le("0.24.1") then
-            io.replace("CMakeLists.txt", " unit_test_framework", "", {plain = true})
-            if package:is_plat("windows") then
-                io.replace("CMakeLists.txt", "if (MSVC)",
-                "if (MSVC)\n    target_link_libraries(${PROJECT_NAME} crypt32)", {plain = true})
-            elseif package:is_plat("mingw") then
-                io.replace("CMakeLists.txt", "if(MINGW)",
-                "if (MINGW)\n    target_link_libraries(${PROJECT_NAME} crypt32)", {plain = true})
-            end
+        io.replace("CMakeLists.txt", " unit_test_framework", "", {plain = true})
+        if package:is_plat("windows") then
+            io.replace("CMakeLists.txt", "if (MSVC)",
+            "if (MSVC)\n    target_link_libraries(${PROJECT_NAME} crypt32)", {plain = true})
+        elseif package:is_plat("mingw") then
+            io.replace("CMakeLists.txt", "if(MINGW)",
+            "if (MINGW)\n    target_link_libraries(${PROJECT_NAME} crypt32)", {plain = true})
         end
 
         local configs = {
@@ -48,7 +45,7 @@ package("mailio")
         }
         table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:is_debug() and "Debug" or "Release"))
         table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
-        if version and version:le("0.23.0") then
+        if package:version() and package:version():le("0.23.0") then
             table.insert(configs, "-DMAILIO_BUILD_SHARED_LIBRARY=" .. (package:config("shared") and "ON" or "OFF"))
         end
         import("package.tools.cmake").install(package, configs)
@@ -57,10 +54,8 @@ package("mailio")
     on_test(function (package)
         assert(package:check_cxxsnippets({test = [[
             #include <mailio/message.hpp>
-            using namespace mailio;
             void test() {
-                message msg;
-                msg.header_codec(message::header_codec_t::QUOTED_PRINTABLE);
+                mailio::message msg;
             }
         ]]}, {configs = {languages = "c++17"}}))
     end)

--- a/packages/m/mailio/xmake.lua
+++ b/packages/m/mailio/xmake.lua
@@ -27,15 +27,9 @@ package("mailio")
     on_install("!iphoneos and !wasm", function (package)
         io.replace("CMakeLists.txt", "/WX", "", {plain = true})
         io.replace("CMakeLists.txt", "set(Boost_USE_STATIC_LIBS ON)", "", {plain = true})
-
         io.replace("CMakeLists.txt", " unit_test_framework", "", {plain = true})
-        if package:is_plat("windows") then
-            io.replace("CMakeLists.txt", "if (MSVC)",
-            "if (MSVC)\n    target_link_libraries(${PROJECT_NAME} crypt32)", {plain = true})
-        elseif package:is_plat("mingw") then
-            io.replace("CMakeLists.txt", "if(MINGW)",
-            "if (MINGW)\n    target_link_libraries(${PROJECT_NAME} crypt32)", {plain = true})
-        end
+        io.replace("CMakeLists.txt", "if (MSVC)", "if (MSVC)\n    target_link_libraries(${PROJECT_NAME} crypt32)", {plain = true})
+        io.replace("CMakeLists.txt", "\nif(MINGW)", "\nif (MINGW)\n    target_link_libraries(${PROJECT_NAME} crypt32)", {plain = true})
 
         local configs = {
             "-DMAILIO_BUILD_EXAMPLES=OFF",

